blogging/views.py

```diff
@@ -0,0 +1,144 @@
+from django.shortcuts import render, redirect
+from django.http import HttpResponse, HttpResponseRedirect, Http404
+from django.template import loader
+from django.utils import timezone
+from blogging.models import Post, Category
+from blogging.forms import PostForm, PublishForm, ListForm
+
+
+def list_view(request):
+    """
+    Displays a list view of each published post.
+
+    In the real world I'd likely want to limit the number of posts displayed incase
+    the blog grows large, with a 'display more' at the bottom.
+    Don't know how to do that yet though...
+    """
+    if request.method == "POST":
+        if not request.user.is_authenticated:
+            return redirect("/login")
+
+        form = PublishForm(request.POST)
+
+        if form.is_valid():
+            post = Post.objects.get(id=form.cleaned_data["post_id"])
+
+            if request.user == post.author:
+                post.published_date = None
+                post.save()
+                return redirect("/")
+        else:
+            print(form.errors)
+
+    form = PublishForm()
+
+    published = Post.objects.exclude(published_date__exact=None).order_by(
```

Comment:
this is where you would iterate through published an update via `pytz` lib (pip install needed) to use system local time for the dates.